corpdir-srv-jpa/src/main/java/com/github/bordertech/corpdir/jpa/common/svc/JpaBasicKeyIdService.java

```diff
@@ -1,19 +0,0 @@
-package com.github.bordertech.corpdir.jpa.common.svc;
-
-import com.github.bordertech.corpdir.api.common.ApiKeyIdObject;
-import com.github.bordertech.corpdir.api.service.BasicKeyIdService;
-import com.github.bordertech.corpdir.jpa.common.feature.PersistKeyIdObject;
-import javax.inject.Singleton;
-
-/**
- * Keyed Entity JPA service implementation.
- *
- * @param <A> the API object type
- * @param <P> the entity type
- * @author Jonathan Austin
- * @since 1.0.0
- */
-@Singleton
-public abstract class JpaBasicKeyIdService<A extends ApiKeyIdObject, P extends PersistKeyIdObject> extends JpaBasicIdService<A, P> implements BasicKeyIdService<A> {
```

Comment:
The Split of the interfaces into `readonly` and `write` is looking good. This will be good for your `sync` module.

To minimise the impact, and being consistent, what do you think of doing the same thing with the `JPA` classes as the `REST` resources by making the `JPA` interfaces implement both the `Readonly` and `Write` interfaces. You could also create a new `CRUD` interface for each entity that implements both `Readonly` and `Write` and use this on the `dataapi`.

This still does not change the `sync` module from having a `Read` for `From` and `Read/Write` for `To`.

Sorry this means winding back a lot of your codes changes for JPA and DATAAPis. Please let me know what you think.

Reply:
Yes I already split all `JPA` services into Read and Write. For example [JpaBasicIdReadOnlyService](https://github.com/kandula-aswin/corporate-directory-server/blob/56b2385e505aea4996e17581484d37ccba0e82e9/corpdir-srv-jpa/src/main/java/com/github/bordertech/corpdir/jpa/readonly/common/svc/JpaBasicIdReadOnlyService.java#L27) and [JpaBasicIdWriteService](https://github.com/kandula-aswin/corporate-directory-server/blob/56b2385e505aea4996e17581484d37ccba0e82e9/corpdir-srv-jpa/src/main/java/com/github/bordertech/corpdir/jpa/modify/common/svc/JpaBasicIdWriteService.java#L23) and so on, see [read jpa services](https://github.com/kandula-aswin/corporate-directory-server/blob/56b2385e505aea4996e17581484d37ccba0e82e9/corpdir-srv-jpa/src/main/java/com/github/bordertech/corpdir/jpa/readonly/common/svc) and [write jpa services](https://github.com/kandula-aswin/corporate-directory-server/blob/56b2385e505aea4996e17581484d37ccba0e82e9/corpdir-srv-jpa/src/main/java/com/github/bordertech/corpdir/jpa/modify/common/svc).

The `REST` and `DataApi` layers brings the read and write interfaces together, see [DefaultCorpCrudVersionDataApi](https://github.com/kandula-aswin/corporate-directory-server/blob/56b2385e505aea4996e17581484d37ccba0e82e9/corpdir-srv-web-ui/src/main/java/com/github/bordertech/corpdir/web/ui/flux/dataapi/DefaultCorpCrudVersionDataApi.java#L19) it is backed by `BasicVersionKeyIdReadOnlyService` and `BasicVersionKeyIdWriteService`. Here is how they look see [ContactResource](https://github.com/kandula-aswin/corporate-directory-server/blob/56b2385e505aea4996e17581484d37ccba0e82e9/corpdir-srv-web-api/src/main/java/com/github/bordertech/corpdir/web/api/resource/v1/ContactResource.java#L28) and [ContactApi](https://github.com/kandula-aswin/corporate-directory-server/blob/56b2385e505aea4996e17581484d37ccba0e82e9/corpdir-srv-web-ui/src/main/java/com/github/bordertech/corpdir/web/ui/flux/dataapi/impl/ContactApi.java#L14).

Hope I catched your point.